src/test/java/com/exasol/validation/ValidationReportTest.java

```diff
@@ -0,0 +1,47 @@
+package com.exasol.validation;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+// [utest->dsn~rr-creates-validation-report~1]
+class ValidationReportTest {
+    private ValidationReport report;
+
+    @BeforeEach
+    void beforeEach() {
+        this.report = new ValidationReport();
+    }
+
+    @Test
+    void testGetFailedValidations() {
+        this.report.addSuccessfulValidation("Pom file.");
+        this.report.addFailedValidations("Some code", "Please, update the version.");
+        assertThat(this.report.getFailedValidations(),
+                containsString("Fail.    Some code: Please, update the version."));
+    }
+
+    @Test
+    void testHasFailedValidations() {
+        assertThat(this.report.hasFailedValidations(), equalTo(false));
+        this.report.addSuccessfulValidation("Pom file.");
+        assertThat(this.report.hasFailedValidations(), equalTo(false));
+        this.report.addFailedValidations("Some code", "Please, update the version.");
+        assertThat(this.report.hasFailedValidations(), equalTo(true));
+        this.report.addSuccessfulValidation("Changes file.");
+        assertThat(this.report.hasFailedValidations(), equalTo(true));
+    }
```

Comment:
@redcatbear does not like multiple asserts without assertAll...
Maybe you could refactor it to a parametrozed test with method source

Reply:
I know, but it's the only case when there is no easy way around. The parameterized test will destroy the readability